compiler/plc_driver/src/lib.rs

```diff
@@ -251,10 +250,7 @@ fn generate_to_string_internal<T: SourceContainer>(
     }
     let module = project.generate_single_module(&context, &options)?;
 
-    module.map(|it| it.persist_to_string()).ok_or_else(|| Diagnostic::GeneralError {
-        message: "Cannot generate module".to_string(),
-        err_no: plc_diagnostics::errno::ErrNo::general__err,
-    })
+    module.map(|it| it.persist_to_string()).ok_or_else(|| Diagnostic::error("Cannot generate module"))
```

Comment:
So initially this was still the enum which defaulted to E001 (Generic Error)
I will for the first step default to E001 so that i don't change all calls  but I want to create some sort of macro that fails if the error is not defined now sure how to do that because yet but it would be a nice to have addition. next step though. For now I think just defaulting to E001 is enough